joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -45,6 +46,47 @@ trait AstForDeclarationsCreator(implicit withSchemaValidation: ValidationMode) {
     Seq(typeDeclAst)
   }
 
+  protected def astForRecordDeclaration(recordDecl: DotNetNodeInfo): Seq[Ast] = {
+    val name     = nameFromNode(recordDecl)
+    val fullName = astFullName(recordDecl)
+    val typeDecl = typeDeclNode(recordDecl, name, fullName, relativeFileName, code(recordDecl))
+    scope.pushNewScope(TypeScope(fullName))
+    val modifiers         = astForModifiers(recordDecl)
+    val paramList         = recordDecl.json(ParserKeys.ParameterList)
+    var membersFromParams = Seq[Ast]()
+
+    // Covers the case where record type can be declared as `record Person(string Name);`
+    // Here, Person should be a TypeDecl and Name should be a member instead of a parameter
+    if (!paramList.isNull) {
```

Comment:
Could we perhaps instead do something like
```scala
val membersFromParams = Try { 
  recordDecl.json(ParserKeys.ParameterList)(ParserKeys.Parameters)
    .arr
    .map(createDotNetNodeInfo)
    .toSeq
  }
 .toOption
 .getOrElse(Seq.empty)
 .map { ... }
 ```
You can also try simply getting a return from the `if` statement instead of prompting an exception, but I would avoid using `var`